smart-contracts/contracts/utils/LockSerializer.sol

```diff
@@ -0,0 +1,156 @@
+
+    // fees
+    uint256 freeTrialLength;
+    uint256 refundPenaltyBasisPoints;
+    uint256 transferFeeBasisPoints;
+
+    // metadata
+    string name;
+    string symbol;
+    // string baseTokenURI; // private?
+    
+    // protocol
+    uint publicLockVersion;
+    address tokenAddress;
+
+    // ownerhsip
+    uint numberOfOwners;
+    uint256 totalSupply; 
+    address[] keyOwners;
+    address[] keyManagers;
+    uint[] expirationTimestamps;
+    // address lockCreator;
+  }
+
+  // We split in multiple structs to avoid "Stack To Deep" solc
+  struct LockPriceInfo {
+    // fees
+    uint expirationDuration;
+    uint keyPrice;
+    uint maxNumberOfKeys;
+    address beneficiary;
+  }
+  
+  struct LockFees {
+    // fees
+    uint256 freeTrialLength;
+    uint256 refundPenaltyBasisPoints;
+    uint256 transferFeeBasisPoints;
+  }
+
+  struct LockMetadata {
+    // metadata
+    string name;
+    string symbol;
+    // string baseTokenURI; // private?
+  }
+
+  function serializePriceInfo(IPublicLockV8 lock) public view returns (LockPriceInfo memory) {
+    uint expirationDuration = lock.expirationDuration();
+    uint keyPrice = lock.keyPrice();
+    uint maxNumberOfKeys = lock.maxNumberOfKeys();
+    address beneficiary = lock.beneficiary();
+    return LockPriceInfo(
+      expirationDuration,
+      keyPrice,
+      maxNumberOfKeys,
+      beneficiary
+    );
+  }
+  
+  function serializeFees(IPublicLockV8 lock) public view returns (LockFees memory) {
+    uint256 freeTrialLength = lock.freeTrialLength();
+    uint256 refundPenaltyBasisPoints = lock.refundPenaltyBasisPoints();
+    uint256 transferFeeBasisPoints = lock.transferFeeBasisPoints();
+    return LockFees(
+      freeTrialLength,
+      refundPenaltyBasisPoints,
+      transferFeeBasisPoints
+    );
+  }
+  
+  function serializeMetadata(IPublicLockV8 lock) public view returns (LockMetadata memory) {
+    string memory name = lock.name();
+    string memory symbol = lock.symbol();
+    return LockMetadata(
+      name,
+      symbol
+    );
+  }
+
+  function serialize(address lockAddress) public view returns (Lock memory) {
+
+    IPublicLockV8 lock = IPublicLockV8(lockAddress); 
+    require( lock.isAlive() == true, "Disabled lock can not be serialized");
+    
+    LockMetadata memory metadata = serializeMetadata(lock);
+    LockFees memory fees = serializeFees(lock);
+    LockPriceInfo memory priceInfo = serializePriceInfo(lock);
+
+    // protocol
+    uint publicLockVersion = lock.publicLockVersion();
+    address tokenAddress = lock.tokenAddress();
+
+    // ownership
+    uint256 totalSupply = lock.totalSupply();
+    uint numberOfOwners = lock.numberOfOwners();
+    
+    // keys
+    address[] memory keyOwners = new address[](totalSupply);
+    address[] memory keyManagers = new address[](totalSupply);
+    uint[] memory expirationTimestamps = new uint[](totalSupply);
+    
+    for (uint256 tokenId = 0; tokenId < totalSupply; tokenId++) {
```

Comment:
to list all key owners we just loop through the whole token supply - so no need for the deprec `getOwnersByPage`